hudi-common/src/main/java/org/apache/hudi/metadata/HoodieBackedTableMetadata.java

```diff
@@ -143,6 +143,13 @@ protected Option<HoodieRecord<HoodieMetadataPayload>> getRecordByKey(String key,
     return recordsByKeys.size() == 0 ? Option.empty() : recordsByKeys.get(0).getValue();
   }
 
+  @Override
+  public List<String> getPartitionPathsWithPrefix(String prefix) throws IOException {
+    return getAllPartitionPaths().stream()
```

Comment:
This is going to be incredibly wasteful in case of MT due to HUDI-4911

Reply:
Yeah, I only put a naive implementation for MDT here. Do you think the lazy fetching is necessary for MDT case? Or is there API to load partitions with given prefix only when using MDT?

Reply:
We should instead fetch for all the prefixes at once, instead of doing it in a for loop

Reply:
Do you mean API like `getPartitionPathsWithPrefix(List<String> prefixes)`

Reply:
Correct